src/Components/Web/src/DynamicRootComponentConfiguration.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using Microsoft.AspNetCore.Components.Web.Infrastructure;
+using static Microsoft.AspNetCore.Internal.LinkerFlags;
+
+namespace Microsoft.AspNetCore.Components.Web
+{
+    /// <summary>
+    /// Configures options for allowing JavaScript to add root components dynamically.
+    /// </summary>
+    public class DynamicRootComponentConfiguration
```

Comment:
```suggestion
    public class JSRootComponentConfiguration
```

Reply:
I think we settled on saying `JSComponent` rather than `JSRootComponent`